pkg/model/dev.go

```diff
@@ -156,15 +169,50 @@ func (b *BuildInfo) GetDockerfilePath() string {
 	return joinPath
 }
 
-// AddArgs add a set of args to the build information
-func (b *BuildInfo) AddArgs(args map[string]string) {
-	for k, v := range args {
-		b.Args = append(b.Args, EnvVar{
+// ExpandBuildArgs add a set of args to the build information
+func (b *BuildInfo) ExpandBuildArgs(previousImageArgs map[string]string) error {
+	if err := b.expandManifestBuildArgs(previousImageArgs); err != nil {
+		return err
+	}
+	return b.expandPreviousImageArgs(previousImageArgs)
+}
+
+func (b *BuildInfo) expandManifestBuildArgs(previousImageArgs map[string]string) error {
+	var err error
+	for idx, arg := range b.Args {
+		if val, ok := previousImageArgs[arg.Name]; ok {
+			oktetoLog.Infof("overriding '%s' with the content of previous build", arg.Name)
+			arg.Value = val
+		}
+		arg.Value, err = ExpandEnv(arg.Value, true)
+		if err != nil {
+			return err
+		}
+		b.Args[idx] = arg
+	}
+	return nil
+}
+
+func (b *BuildInfo) expandPreviousImageArgs(previousImageArgs map[string]string) error {
```

Comment:
```suggestion
func (b *BuildInfo) addExpandedPreviousImageArgs(previousImageArgs map[string]string) error {
```

What do you think about renaming the function? This is actually adding new build args if needed